### ISSUE TITLE
Скварыць спіс палёў, абавязковых для запаўнення

### ISSUE THREAD

**Issue by Olya147** [diglabby/doika#120]
Environment: google chrome.

Steps of reproduce:
1. Открыть модуль Campaigns;
2. Нажать кнопку Create campaign;
3. Нажать на кнопку Save.

Actual result: обязательные поля никак не обозначены. В случае, когда они остались незаполненными, отсутствует сообщение об ошибке.

Expected result: поля, обязательные для заполнения, отмечены *.

**Comment by alies-dev** [comment]
Я дадаў валідацыю на баку бэкэнда, але ці трэба каб апісанне кампаніі было абавязковым полем?

@Olya147 
Было б крута каб та такіх issue мы прыкладалі і спасылку на RTB каб упэўніцца што абавязковае а што апцыянальнае

Дызайн: https://miro.com/app/board/o9J_k0X88dM=/?moveToWidget=3074457346294415052

**Comment by rizomaa** [comment]
@lptn дзякуй, зараз як разумею у нас хаатычна рэализавана абавязковасць палёў. так?

абавязковасць вызначана толькі для увядзення дадзеных ахвярадаўцы.


астатняе сапраўды трэба прапрацаваць.

@nastyafrosty @fr0zen @Olya147 можна абмеркаваць абавязковыя палі...

@lptn пакуль не заганяйцеся, мы зробім спіс усіх палёў і пазначым іх абавязковасць. усё тут акей. не дадасць шмат валакіты з кодам пасля?

**Comment by alies-dev** [comment]
@rizomaa 
>зараз як разумею у нас хаатычна рэализавана абавязковасць палёў. так?

так, але энтрапія невялікая, хаця і яна стварае недавер да ўсіх астатніх частак дызайну

>не дадасць шмат валакіты з кодам пасля?

не, абмяркоўваеце -- "я пачакаю" (с)
![image](https://user-images.githubusercontent.com/5278175/57944018-52477b80-78de-11e9-92ec-38d41f17ddf0.png)

**Comment by rizomaa** [comment]
@lptn па рэкурэнтным там усё зразумела

**Comment by fr0zen** [comment]
@rizomaa тэму трэба будзе абмазгаваць, але таксама без фідбэка гэта будуць гаданні на кафейнай гушчы мне падаецца.

**Comment by rizomaa** [comment]
@lptn тады усе палі неабавязковыя, пасля можна будзе зрабіць таску і прайсція паняць статус. так лепш?

**Comment by alies-dev** [comment]
@rizomaa 
Ну не зусім добра ) На "ўсе" я не згодны, трэьа хоць штосці пакінуць абавязковым, лепейнаадварот зрабіць амаль усе абавязковым а пасля паслабляць патрабаванні і дапісваць ifы ў кодзе

**Comment by fr0zen** [comment]
@fr0zen @lptn вырашыць, якія палі пакідаем абавязковымі, а якія не. Зараз палі апцыянальныя.

**Comment by rizomaa** [comment]
@fr0zen я не зусім разумею, чаму зачыніў задачу. 
Яна актуальная.

**Comment by nastyafrosty** [comment]
@lptn @rizomaa @fr0zen может имейл будет обязательным? во всяком случае при рекурентном он бы нужен был, чтобы приходило уведомление об оплате и можно было отписаться

**Comment by rizomaa** [comment]
> @lptn @rizomaa @fr0zen может имейл будет обязательным? во всяком случае при рекурентном он бы нужен был, чтобы приходило уведомление об оплате и можно было отписаться

@nastyafrosty https://github.com/diglabby/doika/issues/217#issue-408712732

**Comment by rizomaa** [comment]
Абавязковыя палі для кліентскай часткі: 
1. Сума
2. Абранне спосабу ахвяравання.
3. Згода з умовамі арганізацыі па праняццю данэйтаў
4. На другім вакне: у залежнасці ад спосабу, калі абраны рэкурэнтныя плацяжы
5. Абавязковае поле мэйл
6. Усе астатнія палі апцыянальныя і тым ліку на разавых плацяжах

**Comment by rizomaa** [comment]
звярнуся, пасля вырашэння фундаментальнага пытання па таму, як уладкуем віджэт
